diff --git a/MdeModulePkg/MdeModulePkg.dec b/MdeModulePkg/MdeModulePkg.dec
index 3698caf..16201f1 100644
--- a/MdeModulePkg/MdeModulePkg.dec
+++ b/MdeModulePkg/MdeModulePkg.dec
@@ -1884,7 +1884,7 @@
 
   ## Null-terminated Unicode string of the firmware vendor name that is the default name filled into the EFI System Table.
   # @Prompt Firmware vendor.
-  gEfiMdeModulePkgTokenSpaceGuid.PcdFirmwareVendor|L"EDK II"|VOID*|0x00010050
+  gEfiMdeModulePkgTokenSpaceGuid.PcdFirmwareVendor|L"Intel Corporation"|VOID*|0x00010050
 
   ## Firmware revision that is the default revision filled into the EFI System Table.
   # @Prompt Firmware revision.
@@ -2120,7 +2120,7 @@
   #  The OEM assigns each dissimilar table a new OEM Table ID.
   #  This PCD is ignored for definition block.
   # @Prompt Default OEM Table ID for ACPI table creation.
-  gEfiMdeModulePkgTokenSpaceGuid.PcdAcpiDefaultOemTableId|0x20202020324B4445|UINT64|0x30001035
+  gEfiMdeModulePkgTokenSpaceGuid.PcdAcpiDefaultOemTableId|0x2020207076525F55|UINT64|0x30001035
 
   ## Default OEM Revision for ACPI table creation.
   #  According to ACPI specification, for LoadTable() opcode, the OS can also
diff --git a/OvmfPkg/Bhyve/AcpiTables/Dsdt.asl b/OvmfPkg/Bhyve/AcpiTables/Dsdt.asl
index 17e2cf6..d5399e2 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Dsdt.asl
+++ b/OvmfPkg/Bhyve/AcpiTables/Dsdt.asl
@@ -22,7 +22,7 @@
  *     Compiler ID      "INTL"
  *     Compiler Version 0x20150204 (538247684)
  */
-DefinitionBlock ("DSDT.aml", "DSDT", 2, "BHYVE", "BVDSDT", 0x00000001)
+DefinitionBlock ("DSDT.aml", "DSDT", 2, "INTEL ", "U Rvp   ", 0x00000001)
 {
     Name (_S5, Package (0x02)
     {
diff --git a/OvmfPkg/Bhyve/AcpiTables/Facp.aslc b/OvmfPkg/Bhyve/AcpiTables/Facp.aslc
index b3d703e..aa96ef7 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Facp.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Facp.aslc
@@ -7,7 +7,7 @@
 
 #include "Platform.h"
 
-#define EFI_ACPI_OEM_TABLE_ID     SIGNATURE_64('B','V','F','A','C','P',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID     SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 EFI_ACPI_3_0_FIXED_ACPI_DESCRIPTION_TABLE FACP = {
   {
diff --git a/OvmfPkg/Bhyve/AcpiTables/Hpet.aslc b/OvmfPkg/Bhyve/AcpiTables/Hpet.aslc
index 73bb784..c9fb296 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Hpet.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Hpet.aslc
@@ -10,7 +10,7 @@
 #include "Platform.h"
 
 #define EFI_ACPI_HPET_DESCRIPTION_TABLE_SIGNATURE  SIGNATURE_32('H', 'P', 'E', 'T')
-#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('B','V','H','P','E','T',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 //
 // Ensure proper structure formats
diff --git a/OvmfPkg/Bhyve/AcpiTables/Madt.aslc b/OvmfPkg/Bhyve/AcpiTables/Madt.aslc
index e8e956e..7defe23 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Madt.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Madt.aslc
@@ -15,7 +15,7 @@
 #include <IndustryStandard/Acpi.h>
 #include <Platform.h>
 
-#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('B','V','M','A','D','T',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 //
 // Local APIC address
diff --git a/OvmfPkg/Bhyve/AcpiTables/Mcfg.aslc b/OvmfPkg/Bhyve/AcpiTables/Mcfg.aslc
index 62a98d3..048f043 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Mcfg.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Mcfg.aslc
@@ -7,7 +7,7 @@
 
 #include "Platform.h"
 
-#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('B','V','M','C','F','G',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 #pragma pack(1)
 
diff --git a/OvmfPkg/Bhyve/AcpiTables/Platform.h b/OvmfPkg/Bhyve/AcpiTables/Platform.h
index 4561adc..e515642 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Platform.h
+++ b/OvmfPkg/Bhyve/AcpiTables/Platform.h
@@ -20,9 +20,9 @@
 //
 // ACPI table information used to initialize tables.
 //
-#define EFI_ACPI_OEM_ID            'B','H','Y','V','E',' '  // OEMID 6 bytes long
+#define EFI_ACPI_OEM_ID            'I','N','T','E','L',' '  // OEMID 6 bytes long
 #define EFI_ACPI_OEM_REVISION      0x1
-#define EFI_ACPI_CREATOR_ID        SIGNATURE_32('B','H','Y','V')
+#define EFI_ACPI_CREATOR_ID        SIGNATURE_32('I','N','T','L')
 #define EFI_ACPI_CREATOR_REVISION  0x00000001
 
 #define INT_MODEL        0x01
diff --git a/OvmfPkg/Bhyve/AcpiTables/Spcr.aslc b/OvmfPkg/Bhyve/AcpiTables/Spcr.aslc
index ec5114a..24e1a21 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Spcr.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Spcr.aslc
@@ -7,7 +7,7 @@
 
 #include "Platform.h"
 
-#define EFI_ACPI_OEM_TABLE_ID     SIGNATURE_64('B','V','S','P','C','R',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID     SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 EFI_ACPI_SERIAL_PORT_CONSOLE_REDIRECTION_TABLE SPCR = {
   {
diff --git a/OvmfPkg/Bhyve/BhyveRfbDxe/VbeShim.c b/OvmfPkg/Bhyve/BhyveRfbDxe/VbeShim.c
index ea7a364..b355725 100644
--- a/OvmfPkg/Bhyve/BhyveRfbDxe/VbeShim.c
+++ b/OvmfPkg/Bhyve/BhyveRfbDxe/VbeShim.c
@@ -203,11 +203,11 @@ InstallVbeShim (
   VbeInfo     = &VbeInfoFull->Base;
   Ptr         = VbeInfoFull->Buffer;
 
-  CopyMem (VbeInfo->Signature, "VESA", 4);
+  CopyMem (VbeInfo->Signature, "VUSA", 4);
   VbeInfo->VesaVersion = 0x0200;
 
   VbeInfo->OemNameAddress = (UINT32)SegmentC << 12 | (UINT16)((UINTN)Ptr-SegmentC);
-  CopyMem (Ptr, "FBSD", 5);
+  CopyMem (Ptr, "UEFI", 5);
   Ptr += 5;
 
   VbeInfo->Capabilities = BIT1 | BIT0; // DAC can be switched into 8-bit mode
diff --git a/OvmfPkg/Bhyve/BhyveX64.dsc b/OvmfPkg/Bhyve/BhyveX64.dsc
index 5aab790..2271032 100644
--- a/OvmfPkg/Bhyve/BhyveX64.dsc
+++ b/OvmfPkg/Bhyve/BhyveX64.dsc
@@ -519,7 +519,7 @@
 
   gEfiMdeModulePkgTokenSpaceGuid.PcdConInConnectOnDemand|FALSE
 
-  gEfiMdeModulePkgTokenSpaceGuid.PcdFirmwareVendor|L"BHYVE"
+  gEfiMdeModulePkgTokenSpaceGuid.PcdFirmwareVendor|L"INTEL "
   gUefiOvmfPkgTokenSpaceGuid.PcdDebugIoPort|0x2F8
 
 ################################################################################
diff --git a/OvmfPkg/Bhyve/SmbiosPlatformDxe/SmbiosPlatformDxe.c b/OvmfPkg/Bhyve/SmbiosPlatformDxe/SmbiosPlatformDxe.c
index ba21393..ae21b82 100644
--- a/OvmfPkg/Bhyve/SmbiosPlatformDxe/SmbiosPlatformDxe.c
+++ b/OvmfPkg/Bhyve/SmbiosPlatformDxe/SmbiosPlatformDxe.c
@@ -12,9 +12,9 @@
 #include "SmbiosPlatformDxe.h"
 
 #define TYPE0_STRINGS \
-  "EFI Development Kit II / OVMF\0"     /* Vendor */ \
-  "0.0.0\0"                             /* BiosVersion */ \
-  "02/06/2015\0"                        /* BiosReleaseDate */
+  "Intel Corporation\0"   /* Vendor */ \
+  "1.C0\0"                /* BiosVersion */ \
+  "02/06/2024\0"          /* BiosReleaseDate */
 //
 // Type definition and contents of the default Type 0 SMBIOS table.
 //
diff --git a/OvmfPkg/Include/IndustryStandard/Q35MchIch9.h b/OvmfPkg/Include/IndustryStandard/Q35MchIch9.h
index db828d3..ad00b17 100644
--- a/OvmfPkg/Include/IndustryStandard/Q35MchIch9.h
+++ b/OvmfPkg/Include/IndustryStandard/Q35MchIch9.h
@@ -20,7 +20,7 @@
 //
 // Host Bridge Device ID (DID) value for Q35/MCH
 //
-#define INTEL_Q35_MCH_DEVICE_ID  0x29C0
+#define INTEL_Q35_MCH_DEVICE_ID  0x4641 // Change to 0x1480 for AMD MCH, 0x29C0 for Stock MCH, or 0x4641 for Intel MCH
 
 //
 // B/D/F/Type: 0/0/0/PCI
diff --git a/OvmfPkg/QemuVideoDxe/Driver.c b/OvmfPkg/QemuVideoDxe/Driver.c
index 6feca83..94dfb3e 100644
--- a/OvmfPkg/QemuVideoDxe/Driver.c
+++ b/OvmfPkg/QemuVideoDxe/Driver.c
@@ -41,31 +41,31 @@ QEMU_VIDEO_CARD  gQemuVideoCardList[] = {
     L"Cirrus 5446"
   },{
     PCI_CLASS_DISPLAY_VGA,
-    0x1234,
+    0x8086, // 0x1234,
     0x1111,
     QEMU_VIDEO_BOCHS_MMIO,
     L"QEMU Standard VGA"
   },{
     PCI_CLASS_DISPLAY_OTHER,
-    0x1234,
+    0x8086, // 0x1234,
     0x1111,
     QEMU_VIDEO_BOCHS_MMIO,
     L"QEMU Standard VGA (secondary)"
   },{
     PCI_CLASS_DISPLAY_VGA,
-    0x1b36,
+    0x8086, // 0x1b36,
     0x0100,
     QEMU_VIDEO_BOCHS,
     L"QEMU QXL VGA"
   },{
     PCI_CLASS_DISPLAY_VGA,
-    0x1af4,
-    0x1050,
+    0x8086, // 0x1af4,
+    0x0416, // 0x1050,
     QEMU_VIDEO_BOCHS_MMIO,
     L"QEMU VirtIO VGA"
   },{
     PCI_CLASS_DISPLAY_VGA,
-    0x15ad,
+    0x8086, // 0x15ad,
     0x0405,
     QEMU_VIDEO_VMWARE_SVGA,
     L"QEMU VMWare SVGA"
diff --git a/ShellPkg/ShellPkg.dec b/ShellPkg/ShellPkg.dec
index 18df262..1b26ff0 100644
--- a/ShellPkg/ShellPkg.dec
+++ b/ShellPkg/ShellPkg.dec
@@ -123,7 +123,7 @@
   gEfiShellPkgTokenSpaceGuid.PcdShellScreenLogCount|3|UINT8|0x00000008
 
   ## Unicode string of the shell supplier
-  gEfiShellPkgTokenSpaceGuid.PcdShellSupplier|L"EDK II"|VOID*|0x00000010
+  gEfiShellPkgTokenSpaceGuid.PcdShellSupplier|L"Intel Corporation"|VOID*|0x00000010
 
   ## Do extended decode of USB for determining media type
   gEfiShellPkgTokenSpaceGuid.PcdUsbExtendedDecode|TRUE|BOOLEAN|0x00000011
