diff --git a/MdeModulePkg/MdeModulePkg.dec b/MdeModulePkg/MdeModulePkg.dec
index 0775aa9..2794a62 100644
--- a/MdeModulePkg/MdeModulePkg.dec
+++ b/MdeModulePkg/MdeModulePkg.dec
@@ -1897,7 +1897,7 @@
 
   ## Null-terminated Unicode string of the firmware vendor name that is the default name filled into the EFI System Table.
   # @Prompt Firmware vendor.
-  gEfiMdeModulePkgTokenSpaceGuid.PcdFirmwareVendor|L"EDK II"|VOID*|0x00010050
+  gEfiMdeModulePkgTokenSpaceGuid.PcdFirmwareVendor|L"Intel Corporation"|VOID*|0x00010050
 
   ## Firmware revision that is the default revision filled into the EFI System Table.
   # @Prompt Firmware revision.
@@ -2133,7 +2133,7 @@
   #  The OEM assigns each dissimilar table a new OEM Table ID.
   #  This PCD is ignored for definition block.
   # @Prompt Default OEM Table ID for ACPI table creation.
-  gEfiMdeModulePkgTokenSpaceGuid.PcdAcpiDefaultOemTableId|0x20202020324B4445|UINT64|0x30001035
+  gEfiMdeModulePkgTokenSpaceGuid.PcdAcpiDefaultOemTableId|0x2020207076525F55|UINT64|0x30001035
 
   ## Default OEM Revision for ACPI table creation.
   #  According to ACPI specification, for LoadTable() opcode, the OS can also
diff --git a/OvmfPkg/Bhyve/AcpiTables/Dsdt.asl b/OvmfPkg/Bhyve/AcpiTables/Dsdt.asl
index 17e2cf6..d5399e2 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Dsdt.asl
+++ b/OvmfPkg/Bhyve/AcpiTables/Dsdt.asl
@@ -22,7 +22,7 @@
  *     Compiler ID      "INTL"
  *     Compiler Version 0x20150204 (538247684)
  */
-DefinitionBlock ("DSDT.aml", "DSDT", 2, "BHYVE", "BVDSDT", 0x00000001)
+DefinitionBlock ("DSDT.aml", "DSDT", 2, "INTEL ", "U Rvp   ", 0x00000001)
 {
     Name (_S5, Package (0x02)
     {
diff --git a/OvmfPkg/Bhyve/AcpiTables/Facp.aslc b/OvmfPkg/Bhyve/AcpiTables/Facp.aslc
index b3d703e..aa96ef7 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Facp.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Facp.aslc
@@ -7,7 +7,7 @@
 
 #include "Platform.h"
 
-#define EFI_ACPI_OEM_TABLE_ID     SIGNATURE_64('B','V','F','A','C','P',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID     SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 EFI_ACPI_3_0_FIXED_ACPI_DESCRIPTION_TABLE FACP = {
   {
diff --git a/OvmfPkg/Bhyve/AcpiTables/Hpet.aslc b/OvmfPkg/Bhyve/AcpiTables/Hpet.aslc
index 73bb784..c9fb296 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Hpet.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Hpet.aslc
@@ -10,7 +10,7 @@
 #include "Platform.h"
 
 #define EFI_ACPI_HPET_DESCRIPTION_TABLE_SIGNATURE  SIGNATURE_32('H', 'P', 'E', 'T')
-#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('B','V','H','P','E','T',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 //
 // Ensure proper structure formats
diff --git a/OvmfPkg/Bhyve/AcpiTables/Madt.aslc b/OvmfPkg/Bhyve/AcpiTables/Madt.aslc
index e8e956e..7defe23 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Madt.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Madt.aslc
@@ -15,7 +15,7 @@
 #include <IndustryStandard/Acpi.h>
 #include <Platform.h>
 
-#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('B','V','M','A','D','T',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 //
 // Local APIC address
diff --git a/OvmfPkg/Bhyve/AcpiTables/Mcfg.aslc b/OvmfPkg/Bhyve/AcpiTables/Mcfg.aslc
index 62a98d3..048f043 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Mcfg.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Mcfg.aslc
@@ -7,7 +7,7 @@
 
 #include "Platform.h"
 
-#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('B','V','M','C','F','G',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID                      SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 #pragma pack(1)
 
diff --git a/OvmfPkg/Bhyve/AcpiTables/Platform.h b/OvmfPkg/Bhyve/AcpiTables/Platform.h
index 4561adc..e515642 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Platform.h
+++ b/OvmfPkg/Bhyve/AcpiTables/Platform.h
@@ -20,9 +20,9 @@
 //
 // ACPI table information used to initialize tables.
 //
-#define EFI_ACPI_OEM_ID            'B','H','Y','V','E',' '  // OEMID 6 bytes long
+#define EFI_ACPI_OEM_ID            'I','N','T','E','L',' '  // OEMID 6 bytes long
 #define EFI_ACPI_OEM_REVISION      0x1
-#define EFI_ACPI_CREATOR_ID        SIGNATURE_32('B','H','Y','V')
+#define EFI_ACPI_CREATOR_ID        SIGNATURE_32('I','N','T','L')
 #define EFI_ACPI_CREATOR_REVISION  0x00000001
 
 #define INT_MODEL        0x01
diff --git a/OvmfPkg/Bhyve/AcpiTables/Spcr.aslc b/OvmfPkg/Bhyve/AcpiTables/Spcr.aslc
index ec5114a..24e1a21 100644
--- a/OvmfPkg/Bhyve/AcpiTables/Spcr.aslc
+++ b/OvmfPkg/Bhyve/AcpiTables/Spcr.aslc
@@ -7,7 +7,7 @@
 
 #include "Platform.h"
 
-#define EFI_ACPI_OEM_TABLE_ID     SIGNATURE_64('B','V','S','P','C','R',' ',' ')
+#define EFI_ACPI_OEM_TABLE_ID     SIGNATURE_64('U',' ','R','v','p',' ',' ',' ')
 
 EFI_ACPI_SERIAL_PORT_CONSOLE_REDIRECTION_TABLE SPCR = {
   {
diff --git a/OvmfPkg/Bhyve/BhyveRfbDxe/VbeShim.c b/OvmfPkg/Bhyve/BhyveRfbDxe/VbeShim.c
index ea7a364..b355725 100644
--- a/OvmfPkg/Bhyve/BhyveRfbDxe/VbeShim.c
+++ b/OvmfPkg/Bhyve/BhyveRfbDxe/VbeShim.c
@@ -203,11 +203,11 @@ InstallVbeShim (
   VbeInfo     = &VbeInfoFull->Base;
   Ptr         = VbeInfoFull->Buffer;
 
-  CopyMem (VbeInfo->Signature, "VESA", 4);
+  CopyMem (VbeInfo->Signature, "VUSA", 4);
   VbeInfo->VesaVersion = 0x0200;
 
   VbeInfo->OemNameAddress = (UINT32)SegmentC << 12 | (UINT16)((UINTN)Ptr-SegmentC);
-  CopyMem (Ptr, "FBSD", 5);
+  CopyMem (Ptr, "UEFI", 5);
   Ptr += 5;
 
   VbeInfo->Capabilities = BIT1 | BIT0; // DAC can be switched into 8-bit mode
diff --git a/OvmfPkg/Bhyve/BhyveX64.dsc b/OvmfPkg/Bhyve/BhyveX64.dsc
index 131f16f..6879a2b 100644
--- a/OvmfPkg/Bhyve/BhyveX64.dsc
+++ b/OvmfPkg/Bhyve/BhyveX64.dsc
@@ -519,7 +519,7 @@
 
   gEfiMdeModulePkgTokenSpaceGuid.PcdConInConnectOnDemand|FALSE
 
-  gEfiMdeModulePkgTokenSpaceGuid.PcdFirmwareVendor|L"BHYVE"
+  gEfiMdeModulePkgTokenSpaceGuid.PcdFirmwareVendor|L"Intel Corporation"
   gUefiOvmfPkgTokenSpaceGuid.PcdDebugIoPort|0x2F8
 
 ################################################################################
diff --git a/OvmfPkg/Bhyve/SmbiosPlatformDxe/SmbiosPlatformDxe.c b/OvmfPkg/Bhyve/SmbiosPlatformDxe/SmbiosPlatformDxe.c
index ba21393..eacf8c6 100644
--- a/OvmfPkg/Bhyve/SmbiosPlatformDxe/SmbiosPlatformDxe.c
+++ b/OvmfPkg/Bhyve/SmbiosPlatformDxe/SmbiosPlatformDxe.c
@@ -12,9 +12,9 @@
 #include "SmbiosPlatformDxe.h"
 
 #define TYPE0_STRINGS \
-  "EFI Development Kit II / OVMF\0"     /* Vendor */ \
-  "0.0.0\0"                             /* BiosVersion */ \
-  "02/06/2015\0"                        /* BiosReleaseDate */
+  "EFI Development Kit II / OVMF\0"     /* Vendor */ \        // Modified externally via 'patch_ovmf.sh'
+  "0.0.0\0"                             /* BiosVersion */ \   // Modified externally via 'patch_ovmf.sh'
+  "02/06/2015\0"                        /* BiosReleaseDate */ // Modified externally via 'patch_ovmf.sh'
 //
 // Type definition and contents of the default Type 0 SMBIOS table.
 //
@@ -45,9 +45,15 @@ STATIC CONST OVMF_TYPE0  mOvmfDefaultType0 = {
     },
     {      // BIOSCharacteristicsExtensionBytes[2]
       0,   // BiosReserved
-      0x1C // SystemReserved = VirtualMachineSupported |
-           //                  UefiSpecificationSupported |
-           //                  TargetContentDistributionEnabled
+      0x0D // SystemReserved = Hexadecimal > Binary: 0D = 00001101
+           // Bit 0: BIOS Boot Specification supported - 1 (Yes)
+           // Bit 1: Function key-initiated Network Service boot supported - 0 (No)
+           // Bit 2: Enable Targeted Content Distribution - 1 (Yes)
+           // Bit 3: UEFI Specification is supported - 1 (Yes)
+           // Bit 4: SMBIOS table describes a virtual machine - 0 (No)
+           // Bit 5: Manufacturing mode is supported. - 0 (No)
+           // Bit 6: Manufacturing mode is enabled. - 0 (No)
+           // Bit 7: Reserved for future assignment by this specification. - 0 (No)
     },
     0,     // UINT8                     SystemBiosMajorRelease
     0,     // UINT8                     SystemBiosMinorRelease
diff --git a/OvmfPkg/Include/IndustryStandard/Q35MchIch9.h b/OvmfPkg/Include/IndustryStandard/Q35MchIch9.h
index db828d3..86a9e0b 100644
--- a/OvmfPkg/Include/IndustryStandard/Q35MchIch9.h
+++ b/OvmfPkg/Include/IndustryStandard/Q35MchIch9.h
@@ -20,7 +20,7 @@
 //
 // Host Bridge Device ID (DID) value for Q35/MCH
 //
-#define INTEL_Q35_MCH_DEVICE_ID  0x29C0
+#define INTEL_Q35_MCH_DEVICE_ID  0x4641 // Intel G33/P35/Q35/Q33 Chipset - Memory Controller Hub [A0] | Intel Vendor ID Replacement from 0x29c0 | hw/pci-host/q35.c
 
 //
 // B/D/F/Type: 0/0/0/PCI
diff --git a/OvmfPkg/Include/IndustryStandard/QemuPciBridgeCapabilities.h b/OvmfPkg/Include/IndustryStandard/QemuPciBridgeCapabilities.h
index a70b99f..21b697e 100644
--- a/OvmfPkg/Include/IndustryStandard/QemuPciBridgeCapabilities.h
+++ b/OvmfPkg/Include/IndustryStandard/QemuPciBridgeCapabilities.h
@@ -18,7 +18,7 @@
 // The hints apply to PCI Bridges whose PCI_DEVICE_INDEPENDENT_REGION.VendorId
 // equals the following value.
 //
-#define QEMU_PCI_BRIDGE_VENDOR_ID_REDHAT  0x1B36
+#define QEMU_PCI_BRIDGE_VENDOR_ID_REDHAT  0x8086 // "Red Hat, Inc." | AMD Vendor ID Replacement from 0x1b36
 
 //
 // Common capability header for all hints.
diff --git a/OvmfPkg/Library/QemuBootOrderLib/QemuBootOrderLib.c b/OvmfPkg/Library/QemuBootOrderLib/QemuBootOrderLib.c
index 2fe6ab3..2d697df 100644
--- a/OvmfPkg/Library/QemuBootOrderLib/QemuBootOrderLib.c
+++ b/OvmfPkg/Library/QemuBootOrderLib/QemuBootOrderLib.c
@@ -1794,7 +1794,7 @@ StoreQemuBootOrder (
       UnicodeSPrint (
         VariableName,
         sizeof (VariableName),
-        L"VMMBootOrder%04x",
+        L"BootOrder%04x",
         VariableIndex++
         );
       DEBUG ((DEBUG_INFO, "%a: %s = %s\n", __func__, VariableName, Translated));
diff --git a/OvmfPkg/QemuVideoDxe/Driver.c b/OvmfPkg/QemuVideoDxe/Driver.c
index 6feca83..94dfb3e 100644
--- a/OvmfPkg/QemuVideoDxe/Driver.c
+++ b/OvmfPkg/QemuVideoDxe/Driver.c
@@ -41,31 +41,31 @@ QEMU_VIDEO_CARD  gQemuVideoCardList[] = {
     L"Cirrus 5446"
   },{
     PCI_CLASS_DISPLAY_VGA,
-    0x1234,
+    0x8086, // 0x1234,
     0x1111,
     QEMU_VIDEO_BOCHS_MMIO,
     L"QEMU Standard VGA"
   },{
     PCI_CLASS_DISPLAY_OTHER,
-    0x1234,
+    0x8086, // 0x1234,
     0x1111,
     QEMU_VIDEO_BOCHS_MMIO,
     L"QEMU Standard VGA (secondary)"
   },{
     PCI_CLASS_DISPLAY_VGA,
-    0x1b36,
+    0x8086, // 0x1b36,
     0x0100,
     QEMU_VIDEO_BOCHS,
     L"QEMU QXL VGA"
   },{
     PCI_CLASS_DISPLAY_VGA,
-    0x1af4,
-    0x1050,
+    0x8086, // 0x1af4,
+    0x0416, // 0x1050,
     QEMU_VIDEO_BOCHS_MMIO,
     L"QEMU VirtIO VGA"
   },{
     PCI_CLASS_DISPLAY_VGA,
-    0x15ad,
+    0x8086, // 0x15ad,
     0x0405,
     QEMU_VIDEO_VMWARE_SVGA,
     L"QEMU VMWare SVGA"
diff --git a/OvmfPkg/SmbiosPlatformDxe/SmbiosPlatformDxe.c b/OvmfPkg/SmbiosPlatformDxe/SmbiosPlatformDxe.c
index e929da6..56d68aa 100644
--- a/OvmfPkg/SmbiosPlatformDxe/SmbiosPlatformDxe.c
+++ b/OvmfPkg/SmbiosPlatformDxe/SmbiosPlatformDxe.c
@@ -38,9 +38,15 @@ STATIC CONST SMBIOS_TABLE_TYPE0  mOvmfDefaultType0 = {
   },
   {      // BIOSCharacteristicsExtensionBytes[2]
     0,   // BiosReserved
-    0x1C // SystemReserved = VirtualMachineSupported |
-    //                  UefiSpecificationSupported |
-    //                  TargetContentDistributionEnabled
+    0x0D // SystemReserved = Hexadecimal > Binary: 0D = 00001101
+         // Bit 0: BIOS Boot Specification supported - 1 (Yes)
+         // Bit 1: Function key-initiated Network Service boot supported - 0 (No)
+         // Bit 2: Enable Targeted Content Distribution - 1 (Yes)
+         // Bit 3: UEFI Specification is supported - 1 (Yes)
+         // Bit 4: SMBIOS table describes a virtual machine - 0 (No)
+         // Bit 5: Manufacturing mode is supported. - 0 (No)
+         // Bit 6: Manufacturing mode is enabled. - 0 (No)
+         // Bit 7: Reserved for future assignment by this specification. - 0 (No)
   },
   0,     // UINT8                     SystemBiosMajorRelease
   0,     // UINT8                     SystemBiosMinorRelease
@@ -146,21 +152,21 @@ InstallAllStructures (
     VendStr = (CHAR16 *)FixedPcdGetPtr (PcdFirmwareVendor);
     VendLen = StrLen (VendStr);
     if (VendLen < 3) {
-      VendStr = L"unknown";
+      VendStr = L"unknown"; // Modified externally via 'patch_ovmf.sh'
       VendLen = StrLen (VendStr);
     }
 
     VersStr = (CHAR16 *)FixedPcdGetPtr (PcdFirmwareVersionString);
     VersLen = StrLen (VersStr);
     if (VersLen < 3) {
-      VersStr = L"unknown";
+      VersStr = L"unknown"; // Modified externally via 'patch_ovmf.sh'
       VersLen = StrLen (VersStr);
     }
 
     DateStr = (CHAR16 *)FixedPcdGetPtr (PcdFirmwareReleaseDateString);
     DateLen = StrLen (DateStr);
     if (DateLen < 3) {
-      DateStr = L"02/02/2022";
+      DateStr = L"02/02/2022"; // Modified externally via 'patch_ovmf.sh'
       DateLen = StrLen (DateStr);
     }
 
diff --git a/SecurityPkg/Library/AuthVariableLib/AuthServiceInternal.h b/SecurityPkg/Library/AuthVariableLib/AuthServiceInternal.h
index f7bf771..134705d 100644
--- a/SecurityPkg/Library/AuthVariableLib/AuthServiceInternal.h
+++ b/SecurityPkg/Library/AuthVariableLib/AuthServiceInternal.h
@@ -73,8 +73,8 @@ typedef enum {
 /// | AUTH_CERT_DB_DATA          | <-- Last CERT
 /// +----------------------------+
 ///
-#define EFI_CERT_DB_NAME           L"certdb"
-#define EFI_CERT_DB_VOLATILE_NAME  L"certdbv"
+#define EFI_CERT_DB_NAME           L"dbcert"
+#define EFI_CERT_DB_VOLATILE_NAME  L"dbvcert"
 
 #pragma pack(1)
 typedef struct {
diff --git a/ShellPkg/ShellPkg.dec b/ShellPkg/ShellPkg.dec
index 45954fc..e521908 100644
--- a/ShellPkg/ShellPkg.dec
+++ b/ShellPkg/ShellPkg.dec
@@ -120,7 +120,7 @@
   gEfiShellPkgTokenSpaceGuid.PcdShellScreenLogCount|3|UINT8|0x00000008
 
   ## Unicode string of the shell supplier
-  gEfiShellPkgTokenSpaceGuid.PcdShellSupplier|L"EDK II"|VOID*|0x00000010
+  gEfiShellPkgTokenSpaceGuid.PcdShellSupplier|L"Intel Corporation"|VOID*|0x00000010
 
   ## Do extended decode of USB for determining media type
   gEfiShellPkgTokenSpaceGuid.PcdUsbExtendedDecode|TRUE|BOOLEAN|0x00000011
diff --git a/UefiCpuPkg/Library/CpuExceptionHandlerLib/X64/ExceptionHandlerAsm.nasm b/UefiCpuPkg/Library/CpuExceptionHandlerLib/X64/ExceptionHandlerAsm.nasm
index 3d64ac9..9c3f01a 100644
--- a/UefiCpuPkg/Library/CpuExceptionHandlerLib/X64/ExceptionHandlerAsm.nasm
+++ b/UefiCpuPkg/Library/CpuExceptionHandlerLib/X64/ExceptionHandlerAsm.nasm
@@ -57,7 +57,7 @@ ALIGN   8
 AsmIdtVectorBegin:
 %assign Vector 0
 %rep  256
-    push    strict dword %[Vector] ; This instruction pushes sign-extended 8-byte value on stack
+    push    strict qword %[Vector] ; This instruction pushes sign-extended 8-byte value on stack
     push    rax
 %ifdef NO_ABSOLUTE_RELOCS_IN_TEXT
     mov     rax, strict qword 0    ; mov     rax, ASM_PFX(CommonInterruptEntry)
@@ -70,7 +70,7 @@ AsmIdtVectorBegin:
 AsmIdtVectorEnd:
 
 HookAfterStubHeaderBegin:
-    push    strict dword 0      ; 0 will be fixed
+    push    strict qword 0      ; 0 will be fixed
 VectorNum:
     push    rax
 %ifdef NO_ABSOLUTE_RELOCS_IN_TEXT
